### PR TITLE
engine,plan,execution: string and number literals as logical nodes

### PR DESCRIPTION
**Pull request by MichaHoffmann**
Convert more PromQL AST Expressions into Thanos logical nodes. This is an ongoing effort, this PR only does NumberLiteral and StringLiteral.

### CODE REVIEW

**Review by fpetkovski — APPROVED** [review]
Will this also work for label_replace and label_join?

**Comment by MichaHoffmann** [comment]
> Will this also work for label_replace and label_join?

Yes! After this; all expressions returned from "Plan()" will use the logical nodes from this commit

File: logicalplan/plan.go [hunk 1]
```diff
@@ -136,10 +135,10 @@ func Traverse(expr *parser.Expr, transform func(*parser.Expr)) {
 
 func TraverseBottomUp(parent *parser.Expr, current *parser.Expr, transform func(parent *parser.Expr, node *parser.Expr) bool) bool {
 	switch node := (*current).(type) {
-	case *parser.StringLiteral:
-		return false
-	case *parser.NumberLiteral:
-		return false
+	case *parser.StringLiteral, *StringLiteral:
```
Comment (fpetkovski) [inline logicalplan/plan.go:138]: Maybe not something for now, but I think long term all logical nodes should be value objects instead of pointers.

Comment (MichaHoffmann) [reply logicalplan/plan.go:138]: I agree! Lets do that once we have all logical plan nodes in place!

**Pull request merged**